### PR TITLE
Newsletter 2 go

### PR DESCRIPTION
**Pull request by ultramegatom**
Add new newsletter provider: Newsletter2Go: https://www.newsletter2go.com/

### CODE REVIEW

File: src/DependencyInjection/PimcoreCustomerManagementFrameworkExtension.php [hunk 1]
```diff
@@ -57,6 +61,7 @@ protected function loadInternal(array $config, ContainerBuilder $container)
         }
 
         $this->registerGeneralConfiguration($container, $config['general']);
+        $this->registerAuthServerConfiguration($container, $config['oauth_server']);
```
Comment (fashxp) [inline src/DependencyInjection/PimcoreCustomerManagementFrameworkExtension.php]: is this really newsletter2go related?

File: src/DependencyInjection/PimcoreCustomerManagementFrameworkExtension.php [hunk 2]
```diff
@@ -78,6 +83,11 @@ private function registerGeneralConfiguration(ContainerBuilder $container, array
         $container->setParameter('pimcore_customer_management_framework.general.mailBlackListFile', $config['mailBlackListFile']);
     }
 
+    private function registerAuthServerConfiguration(ContainerBuilder $container, array $config)
```
Comment (fashxp) [inline src/DependencyInjection/PimcoreCustomerManagementFrameworkExtension.php]: is this really newsletter2go related?

File: src/DependencyInjection/PimcoreCustomerManagementFrameworkExtension.php [hunk 3]
```diff
@@ -175,4 +192,24 @@ private function configureGDPRDataProvider(ContainerBuilder $container, array $c
         $container->setParameter('pimcore.gdpr-data-extrator.customers', $config['customer']);
 
     }
+
+
+    public function prepend(ContainerBuilder $container)
```
Comment (fashxp) [inline src/DependencyInjection/PimcoreCustomerManagementFrameworkExtension.php]: is this really newsletter2go related?

File: src/DependencyInjection/PimcoreCustomerManagementFrameworkExtension.php [hunk 4]
```diff
@@ -26,12 +26,16 @@
...
 use Symfony\Component\Config\FileLocator;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Extension\PrependExtensionInterface;
 use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
 use Symfony\Component\HttpKernel\DependencyInjection\ConfigurableExtension;
+use Symfony\Component\Yaml\Exception\ParseException;
+use Symfony\Component\Yaml\Yaml;
 
-class PimcoreCustomerManagementFrameworkExtension extends ConfigurableExtension
+class PimcoreCustomerManagementFrameworkExtension extends ConfigurableExtension implements PrependExtensionInterface
```
Comment (fashxp) [inline src/DependencyInjection/PimcoreCustomerManagementFrameworkExtension.php]: is this really newsletter2go related?

File: src/Newsletter/ProviderHandler/Newsletter2Go.php [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+     * @var array $statusMapping
+     * @var array $reverseStatusMapping
+     */
+    protected $statusMapping;
+    protected $reverseStatusMapping;
+
+    protected $doubleOptInFormCode;
+
+
+    public function __construct($shortcut, $listId, $doubleOptInFormCode, array $statusMapping = [], array $reverseStatusMapping = [], array $mergeFieldMapping = [], array $fieldTransformers = [], Newsletter2Go\Newsletter2GoExportService $exportService)
```
Comment (fashxp) [inline src/Newsletter/ProviderHandler/Newsletter2Go.php]: param with no default value after params with default values ... does not make any sense

**Review by fashxp — CHANGES_REQUESTED** [review]

**Comment by markus-moser** [comment]
As there was no update of this PR since the last 5 years it's outdated and therefore I'll close it.

**Pull request closed without merging**